plugins/source-kafka/pkg/factories/source_factory.go

```diff
@@ -37,18 +38,21 @@ func NewKafkaSourceFactory() types.KafkaSourceFactory {
 }
 
 func NewFakeKafkaSourceFactory(ns string) types.KafkaSourceFactory {
-	fakeParams := client.NewFakeKafkaSourceParams()
-	fakeSourceClient := client.NewFakeKafkaSourceClient(fakeParams, ns)
+
+	fakeSourceClient := client.NewFakeKafkaSourceClient(ns)
+	fakeParams := fakeSourceClient.KafkaSourceParams()
 	return &kafkaClientFactory{
 		kafkaSourceParams: fakeParams,
 		kafkaSourceClient: fakeSourceClient,
 		knSourceFactory:   sourcefactories.NewDefaultKnSourceFactory(),
 	}
 }
 
-func (f *kafkaClientFactory) CreateKafkaSourceClient(namespace string) types.KafkaSourceClient {
+// boilerplate
+
```

Comment:
I would move all boilerplate code to `boilerplate.go`... that’s the point of these files :)

Reply:
done